x-pack/plugins/monitoring/public/application/hooks/use_local_storage.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState } from 'react';
+import { has } from 'lodash';
+
+interface ParsedMonitoringData {
+  [key: string]: any;
+}
+export const useLocalStorage = <Value>(key: string, defaultValue: Value): [Value, Function] => {
+  const localStorageMonitoringKey = 'xpack.monitoring.data';
+  const getMonitoringDataStorage = () => {
+    const monitoringDataStorage = window.localStorage.getItem(localStorageMonitoringKey);
+    let parsedData: ParsedMonitoringData = {};
+    try {
+      parsedData = (monitoringDataStorage && JSON.parse(monitoringDataStorage)) || {};
+    } catch (e) {
+      console.error('Monitoring UI: error parsing locally stored monitoring data', e);
```

Comment:
Should we throw here? Looks like CI doesn't like the console error.

Reply:
Looks like https://github.com/elastic/kibana/pull/113318/files#diff-43853b2ff654052ad3c4b15674d71d5eff642c6de3eb82499c9268b130dda3a2 from @neptunian does this with a throw. Maybe we can merge that first and rebase your PR

Reply:
I had coped this hook to him in Slack before removing that.  If he updates this PR it should be fixed now my PR has gone into master.